### PR TITLE
Add Basic Proxy Capability 

### PR DESCRIPTION
**Pull request by jcostacx**
Please review your own code before asking someone else to review it. This is common courtesy.

### Code Review Checklist

- [ ] Is your code covered by unit tests, integration tests and component tests? Did you follow [SCA testing pyramid](https://checkmarx.sharepoint.com/:p:/s/LumoTeam/EdY2yiaZGIVJpvD4rYP4n1UB5PHBVoc_K-TbI8xY7TAHVg?e=hZqufG)?

- [ ] Is your code aligned with our [best practices and conventions](https://checkmarx.atlassian.net/wiki/spaces/CXLUMO/pages/3149037741/Best+Practices+and+Conventions)?

- [ ] Did you add relevant [logs and metrics](https://checkmarx.atlassian.net/wiki/spaces/CXLUMO/pages/3149037741/Best+Practices+and+Conventions#Logging-and-Metrics)?

- [ ] Did you consider scale and performance?

- [ ] Did you handle exceptions and errors in an informative way?

- [ ] Did you follow [REST API guidelines](https://checkmarx.atlassian.net/wiki/spaces/CXLUMO/pages/1719501069/REST+API+Standards)?

### CODE REVIEW

File: src/main/java/com/checkmarx/sca/communication/AccessControlClient.java [hunk 1]
```diff
@@ -53,7 +58,38 @@ public AccessControlClient(final ScannerHelper scannerHelper) {
...
         _authenticationUrl = authenticationUrl;
-        _httpClient = new OkHttpClient();
+        
+        // >> PS:JC - Proxy params - 05-04-2024
+        String proxyHost = configuration.getProxyHost();
+        int proxyPort = configuration.getProxyPort();
+        String proxyUser = configuration.getProxyUsername();
+        String proxyPass = configuration.getProxyPassword();
+
+        if ((!proxyHost.isEmpty()) && (proxyPort >= 10) && (proxyPort <= 65535)) {
```
Comment (OrShamirCM) [inline src/main/java/com/checkmarx/sca/communication/AccessControlClient.java:68]: I see that you used the same method in 2 different file, maybe you can export it to a proxy class that handles it and call it from the 2 clients

**Review by OrShamirCM — COMMENTED** [review]
Looks good, 
Please remove all comments you left
Please write back if you done manual tests to make sure we have no regressios (without proxy) and with proxy with username/without that it passed

File: src/main/java/com/checkmarx/sca/communication/AccessControlClient.java (hunk 1 above)
Comment (jcostacx) [reply src/main/java/com/checkmarx/sca/communication/AccessControlClient.java]: Done! Added ProxyHttpClient class to handle it for the two clients.

File: src/main/java/com/checkmarx/sca/capability/CapabilityDescriptor.java [hunk 2]
```diff
@@ -51,7 +63,10 @@ public String about() {
 
     @Override
     public List<FormField> formFields() {
-        return Arrays.asList(fieldApiUrl, fieldExpirationTime, fieldSecurityRiskThreshold, fieldAllowedLicenses);
+        // >> PS:JC - Proxy params - 05-04-2024
```
[the remark this answers is not in the document]
Comment (jcostacx) [reply src/main/java/com/checkmarx/sca/capability/CapabilityDescriptor.java]: Done! Comments removed.

**Comment by jcostacx** [comment]
> Looks good, Please remove all comments you left Please write back if you done manual tests to make sure we have no regressios (without proxy) and with proxy with username/without that it passed

**Comment by jcostacx** [comment]
Test

**Comment by jcostacx** [comment]
Looks good, Please remove all comments you left Please write back if you done manual tests to make sure we have no regressios (without proxy) and with proxy with username/without that it passed

Yes, tests were done for all three scenarios:
- No proxy, proxy without authentication, and proxy with authentication.
- Used: docker nexus image "sonatype/nexus3" and simple "CCProxy".
- For the tests re-done today, after adding the new "ProxyHttpClient" class.
- The proxy hits from Nexus as seen here
CCProxy

**Review by OrShamirCM — APPROVED** [review]

**Pull request merged**